### PR TITLE
fixes bugs

### LINKED ISSUE #7902: You can add a bear pelt hat to an already advanced space blanket
~~10~~0% tested

### LINKED ISSUE #7903: If you walk across a campfire as a slime you get the 'you've been set on fire' message despite nothing actually happening to you.
to reproduce
1. be slime
2. walk across lit fireplace.
3. be simultaneously on fire and not on fire at the same time.

### LINKED ISSUE #7920: Getting hungry too fast burns you to death
I went from no hunger indicator to starving and started taking burn damage as if on fire.
1. Eat a lot of hyperizine
2. Drink it down with a lot of anti toxin
3. Marvel as you start burning up after a few minutes when you get hungry from it

### LINKED ISSUE #7932: Flaming B-52 fucking obliterates people
One god damn sip will almost overheat you.

### PR DESCRIPTION
**Pull request by ghost**
fixes #7903 
fixes #7920  (I was unable to reproduce the main issue, but could do vomiting and such.)
fixes #7902 
Fixes #7932

### CODE REVIEW

**Comment by gbasood** [comment]
Doesn't this mean that corgis and other simple animals won't catch fire?

File: code/ATMOSPHERICS/hvac/spaceheater.dm [hunk 1]
```diff
@@ -317,7 +317,7 @@
 
 
 /obj/machinery/space_heater/campfire/Crossed(mob/user as mob)
-	if(istype(user,/mob/living/carbon) && on)
+	if(istype(user,/mob/living/carbon/human) && on)
```
Comment (clusterfack) [inline code/ATMOSPHERICS/hvac/spaceheater.dm]: This is pretty shit senpai

**Comment by clusterfack** [comment]
Technically simple animals could never catch fire

File: code/ATMOSPHERICS/hvac/spaceheater.dm (hunk 1 above)
Comment (ghost) [inline code/ATMOSPHERICS/hvac/spaceheater.dm]: I can't think of a better way to do this.

Comment (Kurfursten) [inline code/ATMOSPHERICS/hvac/spaceheater.dm]: !isslime(user)

Comment (ghost) [inline code/ATMOSPHERICS/hvac/spaceheater.dm]: There are other mobs that can't catch fire though

File: code/modules/clothing/suits/miscellaneous.dm [hunk 2]
```diff
@@ -513,11 +513,11 @@
...
-
+	var/bearpelt = 0
 
 /obj/item/clothing/suit/spaceblanket/attackby(obj/item/W,mob/user)
 	..()
-	if(istype(W,/obj/item/clothing/head/bearpelt))
+	if(istype(W,/obj/item/clothing/head/bearpelt) && !bearpelt)
 		to_chat(user,"<span class='notice'>You add \the [W] to \the [src].</span>")
 		qdel(W)
 		qdel(src)
```
Comment (clusterfack) [inline code/modules/clothing/suits/miscellaneous.dm]: This does qdel(src) before spawning at src.loc, you know qdel sets loc to null right?

Comment (ghost) [inline code/modules/clothing/suits/miscellaneous.dm:523]: Well it equips it anyway

**Pull request merged**